pkg/dockerregistry/server/digestcache.go

```diff
@@ -26,20 +27,36 @@ func newDigestToRepositoryCache(size int) (digestToRepositoryCache, error) {
 	return digestToRepositoryCache{Cache: c}, nil
 }
 
```

Comment:
add a `util.Clock` field to `digestToRepositoryCache`, and default it to `util.RealClock`. Propagate that to the repositoryBuckets (either as args, or as fields). That lets you test TTL eviction in a controlled way in unit tests with a fake clock you control